src/components/Stages/StagesController.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import './style.scss';
+import StageItem from './StageItem';
+import OakDialog from '../../oakui/OakDialog';
+import OakViewResolver from '../../oakui/OakViewResolver';
+import OakView from '../../oakui/OakView';
+import { isEmptyOrSpaces } from '../Utils';
+import { sendMessage, receiveMessage } from '../../events/MessageService';
+import OakPrompt from '../../oakui/OakPrompt';
+import OakText from '../../oakui/OakText';
+import OakButton from '../../oakui/OakButton';
+import { fetchStage, saveStage, deleteStage } from '../../actions/StageActions';
+
+interface Props {
+  match: any;
+  setProfile: Function;
+  profile: any;
+  authorization: any;
+  logout: Function;
+  stage: any;
+  fetchStage: Function;
+  saveStage: Function;
+  deleteStage: Function;
+}
+
+const domain = 'stage';
+
+const StageController = (props: Props) => {
+  const [deleteDialogOpen, setDeleteDialogOpen] = useState(false);
+  const [editDialogOpen, setEditDialogOpen] = useState(false);
+  const [data, setData] = useState({
+    id: undefined,
+    name: '',
+  });
+
+  useEffect(() => {
+    const eventBus = receiveMessage().subscribe(message => {
+      if (message.name === domain && message.signal) {
+        sendMessage('notification', true, {
+          type: 'success',
+          message: `${domain} ${message.data.action}`,
+          duration: 5000,
+        });
+        if (message.data.action !== 'deleted') {
+          setEditDialogOpen(!editDialogOpen);
+        }
+      }
+    });
+    return () => eventBus.unsubscribe();
+  });
+
+  useEffect(() => {
+    if (props.authorization.isAuth) {
+      props.fetchStage(props.match.params.tenant, props.authorization);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (props.authorization.isAuth) {
+      props.fetchStage(props.match.params.tenant, props.authorization);
+    }
+  }, [props.authorization.isAuth]);
+
+  useEffect(() => {
+    setData(props.stage.data);
+  }, [props.stage]);
+
+  useEffect(() => {
+    if (!editDialogOpen) {
+      setData({
+        id: undefined,
+        name: '',
+      });
+    }
+  }, [editDialogOpen]);
+
+  const editStage = stage => {
+    setEditDialogOpen(true);
+    setData({
+      id: stage._id,
+      name: stage.name,
+    });
+  };
+
+  const confirmDeleteStage = id => {
+    if (Object.keys(data).length <= 1) {
```

Comment:
change the condition to `Object.keys(data).length === 1`

- It can never be less than 1
- If it is less than 1 (that is 0), you cannot delete anything and no id is obtainable

Reply:
changed the logic